### ISSUE TITLE
About `Thres_metrics`

### ISSUE THREAD

**Issue by SauryChen** [xy-guo/MVSNet_pytorch#55]
During training, I found that the thres2mm_error is larger than 4mm_error and 4mm_error is larger than 8mm_error, which is counter-intuitive (please refer to the attached figure), since the the depth error less than 8mm should be the largest in the three error metrics. I am wondering whether I missed something when reading the metric codes? Thank you!
<img width="1635" alt="截圖 2023-05-22 22 06 24" src="https://github.com/xy-guo/MVSNet_pytorch/assets/61134581/fd1ff922-414e-475e-b6a2-28a5f74bbf79">

**Comment by maxin0002** [comment]
My understanding is that when the accuracy threshold is low, more deviation points will be judged as errors, so the loss will be relatively large. If the threshold is high, fewer deviation points will be judged as errors, so the loss will be smaller.

**Comment by SauryChen** [comment]
> My understanding is that when the accuracy threshold is low, more deviation points will be judged as errors, so the loss will be relatively large. If the threshold is high, fewer deviation points will be judged as errors, so the loss will be smaller.

Thanks. However, in Thres_metrics, this function returns a mask that when the abs(depth_est - depth_gt) is less than the threshold value, the value is 1, and 0 vice versa. So I think there should be less points that can be counted as 1 if the threshold value is small, and Thres_metrics value should be smaller...

**Comment by maxin0002** [comment]
`def Thres_metrics(depth_est, depth_gt, mask, thres):
    assert isinstance(thres, (int, float))
    depth_est, depth_gt = depth_est[mask], depth_gt[mask]
    errors = torch.abs(depth_est - depth_gt)
    err_mask = errors > thres
    return torch.mean(err_mask.float())`
In fact, in the code implementation, the pixel is only 1 if the absolute value is greater than the threshold, so the fewer points, the greater the loss.

**Comment by SauryChen** [comment]
I see, thanks a lot ٩(˃̶͈̀௰˂̶͈́)و. I checked the original code and found I might mistakenly modified the symbol from > to < after running the code ...
